.github/workflows/run_checks.yml

```diff
@@ -0,0 +1,44 @@
+name: Run Checks
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Update submodules
+      run: git submodule update --init --recursive
+    - name: Build docker image
+      run: |
+        bash firmware/tools/build_docker_image.sh
+        echo "Exited with '$?'"
+    - name: Build the discovery firmware
+      run: |
+        bash firmware/tools/compile.sh DISCO
+        echo "Exited with '$?'"
+    - name: Clean the firmware
+      run: |
+        sudo make clean -C firmware
+        echo "Exited with '$?'"
+    - name: Build the nucleo firmware
+      run: |
+        bash firmware/tools/compile.sh NUCLEO
+        echo "Exited with '$?'"
+
+  format_check:
+    needs: build
+    runs-on: ubuntu-latest
+    if: always()
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Check Code Formatting
+      run: make format_check -C firmware
```

Comment:
What's the purpose of "firmware" here

Reply:
to run it in the firmware folder

Reply:
more specifically, to run the make command in the firmware folder

Reply:
https://ftp.gnu.org/old-gnu/Manuals/make-3.79.1/html_node/make_93.html

Reply:
I see thanks, I was confusing -C with the -c passed to the run_formatter script